src/frontend/flight/mod.rs

```diff
@@ -24,7 +28,14 @@ pub async fn run_flight_server(
         .parse()
         .expect("Error parsing the Arrow Flight listen address");
 
-    let handler = SeafowlFlightHandler::new(context);
+    let flush_interval =
+        Duration::from_secs(context.config.misc.sync_conf.flush_task_interval_s);
+    let lock_timeout =
+        Duration::from_secs(context.config.misc.sync_conf.write_lock_timeout_s);
+    let sync_writer = Arc::new(RwLock::new(SeafowlDataSyncWriter::new(context.clone())));
+    let handler = SeafowlFlightHandler::new(context, sync_writer.clone());
+    tokio::spawn(flush_task(flush_interval, lock_timeout, sync_writer));
```

Comment:
Do we need to worry about properly terminating this when Seafowl ends / restarting if it crashes / should this use the subsystem setup from the top level that Seafowl uses to run the GC task and other frontends?